### PR TITLE
[Functions] Add a per-consumer ID to billing

### PR DESCRIPTION
**Pull request by bolekk**

### CODE REVIEW

File: contracts/src/v0.8/functions/dev/0_0_0/FunctionsBillingRegistry.sol [hunk 1]
```diff
@@ -67,6 +67,8 @@ contract FunctionsBillingRegistry is
   }
   // Note a nonce of 0 indicates an the consumer is not assigned to that subscription.
   mapping(address => mapping(uint64 => uint64)) /* consumer */ /* subscriptionId */ /* nonce */ private s_consumers;
+  mapping(address => uint64) /* consumer */ /* consumer_id */ private s_consumerIds;
+  uint64 private lastConsumerId;
```
Comment (pinebit) [inline contracts/src/v0.8/functions/dev/0_0_0/FunctionsBillingRegistry.sol:71]: Should not this be "consumer nonce"?

File: contracts/src/v0.8/functions/dev/0_0_0/FunctionsBillingRegistry.sol [hunk 2]
```diff
@@ -367,9 +369,10 @@ contract FunctionsBillingRegistry is
     address don,
     address client,
     uint64 subscriptionId,
-    uint64 nonce
+    uint64 nonce,
+    uint64 clientId
```
Comment (pinebit) [inline contracts/src/v0.8/functions/dev/0_0_0/FunctionsBillingRegistry.sol:373]: client vs consumer wording is confusing

File: contracts/src/v0.8/functions/dev/0_0_0/FunctionsBillingRegistry.sol (hunk 1 above)
Comment (bolekk) [reply contracts/src/v0.8/functions/dev/0_0_0/FunctionsBillingRegistry.sol:71]: Technically yes, however s_consumers is already holding a nonce (different one) so it'd be a little confusing.

File: contracts/src/v0.8/functions/dev/0_0_0/FunctionsBillingRegistry.sol (hunk 2 above)
Comment (bolekk) [reply contracts/src/v0.8/functions/dev/0_0_0/FunctionsBillingRegistry.sol:373]: I agree. @justinkaseman OK to reword to consumer everywhere?

**Comment by bolekk** [comment]
not needed for billing v1, closing

**Pull request closed without merging**